gtk/src/toga_gtk/app.py

```diff
@@ -21,8 +27,9 @@ def __init__(self, interface):
         self.interface = interface
         self.interface._impl = self
 
-        gbulb.install(gtk=True)
-        self.loop = asyncio.new_event_loop()
+        self.policy = GLibEventLoopPolicy()
+        asyncio.set_event_loop_policy(self.policy)
+        self.loop = asyncio.get_event_loop()
```

Comment:
Good catch. `new_event_loop()` won't work because it will create the loop, and PyGobject wants to take responsibility for that. If you use `new_event_loop()`, `on_running()` won't fire (or anything else sent to self.loop) won't fire because it's on a different event loop that is never started.

However, we *can* call `get_running_loop()` as soon as the GTK `activate` signal is fired, which is before `startup()` is invoked.

Reply:
> Good catch. `new_event_loop()` won't work because it will create the loop, and PyGobject wants to take responsibility for that. If you use `new_event_loop()`, `on_running()` won't fire (or anything else sent to self.loop) won't fire because it's on a different event loop that is never started.

Looking at the docs for `get_event_loop()`, it says it will call `get_event_loop_policy().get_event_loop()` if there isn't a running event loop....and this is the same behavior as `new_event_loop()`.

Given the policy was just set to `GLibEventLoopPolicy`, wouldn't this still allow PyGObject to own loop creation?

Reply:
Hrm... I can't argue with the logic... but the `handlers` example app isn't starting the `on_running` handler when I switch to `new_event_loop`. Digging into this now...

Reply:
Found the problem. It turns out that the default `get_event_loop()` implementation returns the event loop for the current thread - and that *isn't* the "default" context for the GTK app. However, you can get the loop for the default context fairly easily.

Reply:
Actually - it's even easier than that. The GTK event loop policy implementation of `new_event_loop()` explicitly creates a new Gtk MainContext - which isn't what we want. But `get_event_loop()` *on the policy* will create the event loop in an appropriate context, rather than creating a new one.